document/core/exec/instructions.rst

```diff
@@ -464,6 +998,98 @@ Memory Instructions
    \end{array}
 
 
+.. _exec-load-extend:
+.. _exec-load-splat:
+
+:math:`\V128\K{.}\LOAD{M}\K{x}L\_\sx~\memarg` and :math:`\V128\K{.}\LOAD{N}\K{\_splat}~\memarg`
+...............................................................................................
+
+1. Let :math:`F` be the :ref:`current <exec-notation-textual>` :ref:`frame <syntax-frame>`.
+
+2. Assert: due to :ref:`validation <valid-load-extend>`, :math:`F.\AMODULE.\MIMEMS[0]` exists.
+
+3. Let :math:`a` be the :ref:`memory address <syntax-memaddr>` :math:`F.\AMODULE.\MIMEMS[0]`.
+
+4. Assert: due to :ref:`validation <valid-load-extend>`, :math:`S.\SMEMS[a]` exists.
+
+5. Let :math:`\X{mem}` be the :ref:`memory instance <syntax-meminst>` :math:`S.\SMEMS[a]`.
+
+6. Assert: due to :ref:`validation <valid-load-extend>`, a value of :ref:`value type <syntax-valtype>` |I32| is on the top of the stack.
+
+7. Pop the value :math:`\I32.\CONST~i` from the stack.
+
+8. Let :math:`\X{ea}` be the integer :math:`i + \memarg.\OFFSET`.
+
+9. If :math:`N` is not part of the instruction, then:
+
+   a. Let :math:`N` be :math:`M * L`.
+
+10. If :math:`\X{ea} + N/8` is larger than the length of :math:`\X{mem}.\MIDATA`, then:
```

Comment:
This means that, even though the load-extend instructions read 8 bytes, they'll fail if they can't read 16 bytes. Is that intentional?

EDIT: Ah, n/m. This works because N for extend instructions is actually half the full size.

Reply:
Yea sorry for the confusion, I took your suggestion to split load extend and load splat up, that should make it clearer!